storage/rocksdb/ha_rocksdb.cc

```diff
@@ -4895,13 +4898,16 @@ void ha_rocksdb::setup_read_decoders() {
 }
 
 #ifndef NDEBUG
-void dbug_append_garbage_at_end(std::string &on_disk_rec) {
-  on_disk_rec.append("abc");
+void dbug_append_garbage_at_end(rocksdb::PinnableSlice &on_disk_rec) {
+  on_disk_rec.Reset();
```

Comment:
It seems this doesn't append "abc" to the end of the record, but resets the record to "abc". Could the content of on_disk_rec be copied out first and then added back to the on_disk_rec with "abc" appended to it?

Reply:
Yep, I can do it. Interestingly that all tests passed :)